cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/event/EventBusTest.java

```diff
@@ -162,19 +167,178 @@ public void multipleEvents() throws Exception {
+        bus.register(new ProtectedSubScriber());
+
+        // Post an event
+        TestEvent e = new TestEvent();
+        bus.post(e);
+    }
 
+    @Test
+    public void anonymousClass() throws Exception {
+        thrown.expect(IllegalArgumentException.class);
+        thrown.expectMessage("Event Subscriber class needs to be public");
+        bus.register(new Object(){
             @Subscribe
-            public void throwOnEvent(TestEvent event) throws Exception {
-                throw new Exception("Test subscriber exception");
+            public void doNothing(Object object){
+
             }
         });
 
         // Post an event
         TestEvent e = new TestEvent();
         bus.post(e);
+    }
 
-        // Test passes if no exception is thrown
+    @Test
+    public void packageProtectedSubScriber() throws Exception {
+        thrown.expect(IllegalArgumentException.class);
+        thrown.expectMessage("Event Subscriber class needs to be public");
+
+        bus.register(new PackageProtectedSubscriber());
+
+        // Post an event
+        TestEvent e = new TestEvent();
+        bus.post(e);
+    }
+
+    @Test
+    public void packageProtectedSubMethod(){
+        thrown.expect(IllegalArgumentException.class);
+        thrown.expectMessage("Subscriber method with @Subscribe is required to be public");
+
+        bus.register(new PakageProtectedMethodSubscriber());
+
+        // Post an event
+        TestEvent e = new TestEvent();
+        bus.post(e);
+    }
+
+    @Test
+    public void protectedSubMehtod(){
+        thrown.expect(IllegalArgumentException.class);
+        thrown.expectMessage("Subscriber method with @Subscribe is required to be public");
+        bus.register(new ProtectedMethodSub());
+
+        // Post an event
+        TestEvent e = new TestEvent();
+        bus.post(e);
+    }
+
+    @Test
+    public void privateSubMethod(){
+        thrown.expect(IllegalArgumentException.class);
+        thrown.expectMessage("Subscriber method with @Subscribe is required to be public");
+        bus.register(new PrivateMethodSub());
+
+        // Post an event
+        TestEvent e = new TestEvent();
+        bus.post(e);
+    }
+
+    @Test
+    public void subMethodWith2Parameters(){
+        thrown.expect(IllegalArgumentException.class);
+        thrown.expectMessage("Subscriber method is required to have only 1 parameter");
+
+        bus.register(new SubscriberMethod2Params());
+
+        // Post an event
+        TestEvent e = new TestEvent();
+        bus.post(e);
+    }
+
+    @Test
+    public void subMethodWith0Parameters(){
+        thrown.expect(IllegalArgumentException.class);
+        thrown.expectMessage("Subscriber method is required to have only 1 parameter");
+
+        bus.register(new SubscriberMethod0Params());
+
+        // Post an event
+        TestEvent e = new TestEvent();
+        bus.post(e);
+    }
+
+    public static class SubscriberMethod2Params{
```

Comment:
Can we put all of these statics in one place? There are some existing ones at the top and one in the middle